tests/test_detection.py

```diff
@@ -162,6 +161,77 @@ def test_cat_id_remap(self):
         for k in result.keys():
             self.assertTrue(torch.allclose(result[k], result_remap_cat_id[k]))
 
+    def test_class_wise_metrics(self):
+        PREDICTIONS = [[[0, 1.0, 0, 0, 1, 1],
+                        [1, 1.0, 0, 0, 1, 1]]]
+
+        TARGETS = [[[0, 0, 0, 1, 1],
+                    [1, 0, 0, 0, 0]]]
+
+        metric = MeanAveragePrecision(iou_thresholds=[0.5], class_metrics=True)
+        metric.update(PREDICTIONS, TARGETS)
+        results = metric.compute()
+        self.assertAlmostEqual(results['map_per_class'].get(0), 1.0, places=5)
+        self.assertAlmostEqual(results['map_per_class'].get(1), 0.0, places=5)
+
+    def test_class_wise_with_missing_class_in_predictions(self):
+        PREDICTIONS = [[[1, 1.0, 0, 0, 1, 1]]]
+        TARGETS = [[[1, 0, 0, 1, 1],
+                    [2, 0, 0, 1, 1]]]
+
+        metric = MeanAveragePrecision(iou_thresholds=[0.5], class_metrics=True)
+        metric.update(PREDICTIONS, TARGETS)
+        results = metric.compute()
+        self.assertAlmostEqual(results['map_per_class'].get(1), 1.0, places=5)
+        self.assertAlmostEqual(results['map_per_class'].get(2), 0.0, places=5)
+
+    def test_class_wise_with_missing_class_in_targets(self):
+        PREDICTIONS = [[[1, 1.0, 0, 0, 1, 1],
+                        [2, 1.0, 0, 0, 1, 1]]]
+        TARGETS = [[[2, 0, 0, 1, 1]]]
+
+        metric = MeanAveragePrecision(iou_thresholds=[0.5], class_metrics=True)
+        metric.update(PREDICTIONS, TARGETS)
+        results = metric.compute()
+        self.assertAlmostEqual(results['map_per_class'].get(1), -1.0, places=5)
+        self.assertAlmostEqual(results['map_per_class'].get(2), 1.0, places=5)
+
+    def test_class_wise_after_reset(self):
+        PREDICTIONS = [[[1, 1.0, 0, 0, 1, 1]]]
```

Comment:
Can you have multiple test cases covering different scenarios:

more classes before reset
more classes after reset
..

Wrt info on parametrized tests:
https://chat.openai.com/share/9841c3a6-3be4-4450-b3a5-b6d7cbe81250